src/assignment/hooks/assignment-block-meta.tsx

```diff
@@ -0,0 +1,53 @@
+import { AssignmentBlock } from '@viaa/avo2-types/types/assignment';
+import { useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Link } from 'react-router-dom';
+import { APP_PATH } from '../../constants';
+import { SearchFilter } from '../../search/search.const';
+import { buildLink, formatDate } from '../../shared/helpers';
+
+export function useAssignmentBlockMeta() {
```

Comment:
wouldn't this be better as a component?
```
 <AssignmentBlockMeta>
```

This is how i usually decide between a custom hook and a component:
* if a custom hook only returns one ReactNode it is better suited to be a function component
* if a hook returns multiple variables or the return type if not a ReactNode's then a custom hook makes sense